theano/tensor/basic.py

```diff
@@ -29,6 +29,7 @@
 from theano.printing import pprint, min_informative_str
 # For history
 from theano.compile import Rebroadcast, Shape, shape
+from theano.scalar import Scalar
```

Comment:
Instead of `Scalar`, import `int32` here, which is a predefined Scalar instance for 32-bits integers.

```python
from theano.scalar import int32
```